src/ir/typescript_types.py

```diff
@@ -82,9 +81,17 @@ def get_non_nothing_types(self): # Overwriting Parent method to add TS-specific
         types.extend([
             self.get_null_type(),
             UndefinedType(primitive=False),
+            literal_types.get_literal(),
         ])
         return types
 
+    def constant_candidates(self, gen_object):
+        from src.ir import ast
+        return {
+            "NumberLiteralType": lambda etype: ast.IntegerConstant(etype.name),
```

Comment:
Assuming the following number literal type: "1"

Is the name of the above type `NumberLiteralType` or "1"?. If it is the second case, this code is dead.

Reply:
Name of the type is "1". It is dead. Removed since we don't create actual constants, we just use the already generated names.

Reply:
> we just use the already generated names.
where?

Reply:
For the `StringLiteralType`, get_name actually returns `" + self.name + "` as seen [here](https://github.com/alexisthedev/hephaestus/pull/12/files#diff-04b80108daf0258108951e40850a31fbcdb0f6a68f2b2b80c7b6aac184a63316R239).

For the `NumberLiteralType`, it just returns the name which is converted to a string in the translator (same as all other type names).

Since the name is the identifier for the literal (generated [here](https://github.com/alexisthedev/hephaestus/pull/12/files#diff-04b80108daf0258108951e40850a31fbcdb0f6a68f2b2b80c7b6aac184a63316R267) for string literals and [here](https://github.com/alexisthedev/hephaestus/pull/12/files#diff-04b80108daf0258108951e40850a31fbcdb0f6a68f2b2b80c7b6aac184a63316R282) for number literals) we don't need to create any Constant AST nodes.

Reply:
True. Will get on that.